### PR TITLE
Resolve -From arguments in Script-Migration

### LINKED ISSUE #6126: Migrations: Script-Migration doesn't revert -From migration
In the previous version of EF core one could use Update-Database with the option -Script to generate an SQL Script instead of updating the DB. That way I could generate the script to downgrade a bad migration.
Now the option -script has gone and I'm not sure how to get the same result. 
I tried Script-Migration -From Migration2 -To Migration1 (where Migration2 is newer than Migration1) but no success.
Any hints...?

Thanks

Paolo

### LINKED ISSUE #6228: Migrations: Using name with Script-Migration doesn't work (only ID)
I use Entity Framework Core and PostgreSQL

**project.json**
```
{
    "authors": [ "sam" ],
    "buildOptions": {
        "emitEntryPoint": "true"
    },
    "dependencies": {
        "Microsoft.EntityFrameworkCore": "1.0.0",
        "Microsoft.EntityFrameworkCore.Design": "1.0.0-preview2-final",

        "Npgsql.EntityFrameworkCore.PostgreSQL": "1.0.1",
        "Npgsql.EntityFrameworkCore.PostgreSQL.Design": "1.0.1",

        "Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore": "1.0.0",

        "Microsoft.AspNetCore.Identity.EntityFrameworkCore": "1.0.0",

        "Microsoft.Extensions.Configuration.Abstractions": "1.0.0",
        "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0",
        "Microsoft.Extensions.Configuration.Json": "1.0.0",

        "Microsoft.Extensions.Logging": "1.0.0",
        "Microsoft.Extensions.Logging.Console": "1.0.0",
        "Microsoft.Extensions.Logging.Debug": "1.0.0",

        "DelegateDecompiler": "0.18.1",
        "DelegateDecompiler.EntityFramework": "0.18.1",
    },
    "description": "Xrm.Hcs.DataLayer Class Library",
    "frameworks": {
        "net451": {
            "frameworkAssemblies": {
                "Microsoft.CSharp": "4.0.0.0",
                "System.Collections": "4.0.0.0",
                "System.Linq": "4.0.0.0",
                "System.Threading": "4.0.0.0",
                "System.Runtime": "4.0.10.0",
                "System.ComponentModel.DataAnnotations": "4.0.0.0",
                "System.Data": "4.0.0.0",
                "System.Transactions": "4.0.0.0"
            }
        }
    },
    "tools": {
        "Microsoft.EntityFrameworkCore.Tools": "1.0.0-preview2-final"
    },
    "version": "1.0.0-*"
}
```

**Startup.cs**
```
       public void ConfigureServices(IServiceCollection services)
       {
            services.AddDbContext<MyContext>(options => options.UseNpgsql(Configuration["ConnectionStrings:PgSqlServer"]));
        }
```

**MyContext.cs**
```
    public class HcsContext : DbContext
    {
        public HcsContext(DbContextOptions<HcsContext> contextOptions) : base(contextOptions)
        {
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
            // some configs
            foreach (var fk in modelBuilder.Model.GetEntityTypes().SelectMany(x=>x.GetForeignKeys()))
            {
                fk.DeleteBehavior = DeleteBehavior.Restrict;
            }
        }
    }
```

When I run command:
**dotnet ef migrations script V038Migration V039Migration -o Migrations/Sql.Up/migration_v038_to_v039.sql**
everything is okay: there are no errors. When command finished, it returns "Done".
But **migration_v038_to_v039.sql** is empty.
If I generate *.cs class with migration code - it is not empty (and it has code i expected)

But if I run next command:
**dotnet ef migrations script V039Migration V038Migration -o Migrations/Sql.Down/migration_v039_to_v038.sql**
then **migration_v039_to_v038.sql** will be not empty

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #6126 & fixes #6228

### CODE REVIEW

**Comment by natemcmaster** [comment]
Nit: probably time to rebase. GH is picking up commits not related to the actual PR

File: test/Microsoft.EntityFrameworkCore.Sqlite.FunctionalTests/MigrationsSqliteTest.cs [hunk 1]
```diff
@@ -76,6 +76,34 @@ public override void Can_generate_up_scripts()
...
 
+        public override void Can_generate_one_up_script()
+        {
+            base.Can_generate_one_up_script();
+
+            Assert.Equal(
+                @"ALTER TABLE ""Table1"" RENAME TO ""Table2"";
+
+INSERT INTO ""__EFMigrationsHistory"" (""MigrationId"", ""ProductVersion"")
+VALUES ('00000000000002_Migration2', '7.0.0-test');
```
Comment (natemcmaster) [inline test/Microsoft.EntityFrameworkCore.Sqlite.FunctionalTests/MigrationsSqliteTest.cs]: Nit: EF 7? What's that? :trollface:

**Comment by natemcmaster** [comment]
LGTM. :shipit:

**Comment by bricelam** [comment]
Closing. My *merges into* branch got renamed and it's messed up the PR.

**Comment by natemcmaster** [comment]
If only we had known this was coming ;) https://github.com/blog/2224-change-the-base-branch-of-a-pull-request

**Comment by bricelam** [comment]
Ha! About time...

**Pull request closed without merging**